discovery/kubernetes/metrics.go

```diff
@@ -37,10 +38,18 @@ func newDiscovererMetrics(reg prometheus.Registerer, rmi discovery.RefreshMetric
 			},
 			[]string{"role", "event"},
 		),
+		failuresCount: prometheus.NewCounter(
```

Comment:
naming+description ok?

Reply:
Maybe clarify that not ALL failures will be counted (as some are silently retried: watches 429 e.g.) and prevent users from misusing this counter...

instead of a counter, we could expose a gauge with the timestamp of the last failure occurrence.

WDYT?